File: backend/src/entity/Ponto.ts
import {
  Column,
  PrimaryGeneratedColumn,
  Entity,
  CreateDateColumn,
  UpdateDateColumn,
  OneToMany,
} from "typeorm";

@Entity("pontos")
export class Ponto {
  @PrimaryGeneratedColumn()
  id?: number;

  @Column("varchar", { length: 100, nullable: false })
  name?: string;

  @Column("varchar", { length: 100, nullable: false })
  description?: string;

  @Column("varchar", { length: 100, nullable: false })
  reference?: string;

  @Column("varchar", { length: 2, nullable: false })
  state?: string;

  @Column("varchar", { length: 100, nullable: false })
  city?: string;

  @CreateDateColumn()
  insertedAt?: Date;

  @UpdateDateColumn()
  updatedAt?: Date;
}

File: PontosTuristicosBackend/PontosTuristicosBackend/Model/Ponto.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace PontosTuristicosBackend.Model
{
    public class Ponto
    {
        public int Id { get; set; }
        public String? Name { get; set; }
        public String? Description { get; set; }
        public String? Reference { get; set; }
        public String? State { get; set; }
        public String? City { get; set; }
        public DateTime InsertedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public bool Exists(String? search)
        {
            if (search == null || search == "")
            {
                return true;
            }

            if (Name != null)
            {
                if (Name.Contains(search))
                {
                    return true;
                }
            }

            if (Description != null)
            {
                if (Description.Contains(search))
                {
                    return true;
                }
            }

            if (Reference != null)
            {
                if (Reference.Contains(search))
                {
                    return true;
                }
            }

            if (State != null)
            {
                if (State.Contains(search))
                {
                    return true;
                }
            }

            if (City != null)
            {
                if (City.Contains(search))
                {
                    return true;
                }
            }

            return false;
        }

    }
}